### ISSUE TITLE
Switching criteria with non-monotone interval has difference in paper and code?

### ISSUE THREAD

**Issue by yashu-seth** [salesforce/awd-lstm-lm#73]
Hello Everyone,

I went through the paper [Regularizing and Optimizing LSTM Language Models](https://arxiv.org/abs/1708.02182). In the algorithm (NT-ASGD) discussed in the paper, for the switching to take place, current validation loss should be greater than the last **n** intervals, where **n** is the non-monotone interval.

While in code the implementation suggests that current validation loss should be greater that all but the last **n** intervals.

```python
if args.optimizer == 'sgd' and 't0' not in optimizer.param_groups[0] and
 (len(best_val_loss)>args.nonmono and val_loss > min(best_val_loss[:-args.nonmono])):
                print('Switching to ASGD')
                optimizer = torch.optim.ASGD(model.parameters(), lr=args.lr, t0=0, lambd=0., weight_decay=args.wdecay)
```
I think the code condition should be ``` min(best_val_loss[-args.nonmono:]))``` instead of ``` min(best_val_loss[:-args.nonmono]))``` .

Please correct me if am missing something.

Thanks.

**Comment by angeliand** [comment]
Hi! 
Paper has a typo, instead of taking the last **n**, they are masked and we compare against the remaining set. Check #13 for confirmation. 
Hope this helps.

**Comment by yashu-seth** [comment]
@angeliand Thanks a lot!!
